### PR TITLE
Add Gatling tests to test performance.

### PR DESCRIPTION
**Pull request by cbickel**
Add Gatling tests to test performance.

## Description
This PR adds gatling to the project. It can be used to test the performance of an Openwhisk system.
To easily write new tests, I also provided a small `OpenWhisk-Gatling-extension` and some example tests.

## Related issue and scope
- [ ] I opened an issue to propose and discuss this change

## My changes affect the following components
- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [x] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation

## Types of changes
- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:
- [x] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

**Comment by dubee** [comment]
@cbickel, have you thought about putting these performance tests in https://github.com/apache/incubator-openwhisk-performance?

**Comment by cbickel** [comment]
@dubee I thought about that, but I decided to pull the content of the performance repository here instead. I also wrote a mail to the dev list to hear some other opinions on that.

**Comment by dubee** [comment]
Restarted Travis, unrelated Kafka test failed.

**Comment by dubee** [comment]
@cbickel, closing this PR since it looks like we went with https://github.com/apache/incubator-openwhisk/pull/3526.

**Comment by cbickel** [comment]
@dubee Reopening this PR as WIP, as here is an additional Load Test and the ability to test action-related calls.

File: performance/gatling_tests/src/gatling/scala/BlockingInvokeOneActionSimulation.scala [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import scala.concurrent.duration._
+
+class BlockingInvokeOneActionSimulation extends Simulation {
+  // Specify parameters for the run
+  val host = sys.env("OPENWHISK_HOST")
+
+  // Specify authentication
+  val Array(uuid, key) = sys.env("API_KEY").split(":")
+
+  val connections: Int = sys.env("CONNECTIONS").toInt
```
Comment (markusthoemmes) [inline performance/gatling_tests/src/gatling/scala/BlockingInvokeOneActionSimulation.scala]: No default value here?

File: performance/gatling_tests/src/gatling/scala/BlockingInvokeOneActionSimulation.scala [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  val seconds: FiniteDuration = sys.env.getOrElse("SECONDS", "10").toInt.seconds
+
+  // Specify thresholds
+  val requestsPerSec: Int = sys.env("REQUESTS_PER_SEC").toInt
```
Comment (markusthoemmes) [inline performance/gatling_tests/src/gatling/scala/BlockingInvokeOneActionSimulation.scala]: No default value?

**Pull request merged**